app.rb

```diff
@@ -0,0 +1,132 @@
+    @books.each { |book| puts "[Book] Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def list_of_persons
+    puts 'Empty! Add person.' if @persons.empty?
+    @persons.each { |person| puts "[#{person.class.name}] Name: #{person.name}, Age: #{person.age}, id: #{person.id}" }
+  end
+
+  def create_person
+    print 'To create a student, press 1, To create a teacher, press 2 : '
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid. Try again'
+    end
+  end
+
+  def create_student
+    puts 'Create new student'
+    print 'Enter student age: '
+    age = gets.chomp.to_i
+    print 'Enter name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+    case parent_permission
+    when 'n'
+      student = Student.new('classroom', age, name, parent_permission: false)
+      @persons << student
+      puts 'Student doesnt have parent permission, cant rent books'
+    when 'y'
+      student = Student.new('classroom', age, name, parent_permission: true)
+      @persons << student
+      puts 'Student created successfully'
+    end
+  end
+
+  def create_teacher
+    puts 'Create a new teacher'
+    print 'Enter teacher age: '
+    age = gets.chomp.to_i
+    print 'Enter teacher name: '
+    name = gets.chomp
+    print 'Enter teacher specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(specialization, age, name)
+    @persons << teacher
+    puts 'Teacher created successfully'
+  end
+
+  def create_book()
+    puts 'Create a new book'
+    print 'Enter title: '
+    title = gets.chomp
+    print 'Enter author: '
+    author = gets
+    book = Book.new(title, author)
+    @books.push(book)
+    puts "Book #{title} created successfully."
+  end
+
+  def create_rental
+    puts 'Select which book you want to rent by entering its number'
+    @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+
+    book_id = gets.chomp.to_i
+
+    puts 'Select a person from the list by its number'
+    @persons.each_with_index do |person, index|
+      puts "#{index}) [#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+
+    person_id = gets.chomp.to_i
+
+    print 'Date: '
+    date = gets.chomp.to_s
+
+    rental = Rental.new(date, @books[book_id], @persons[person_id])
+    @rentals << rental
+
+    puts 'Rental created successfully'
+  end
+
+  def list_of_rentals
+    print 'To see rentals enter the person ID: '
+    id = gets.chomp.to_i
+
+    puts 'Rented Books:'
+    @rentals.each do |rental|
+      if rental.person.id == id
+        puts "Peson: #{rental.person.name}  Date: #{rental.date}, Book: '#{rental.book.title}' by #{rental.book.author}"
+      else
+        puts
+        puts 'No records where found for the given ID'
+      end
+    end
+  end
+end
```

Comment:
- [ ] (Optional) Great job implementing the changes suggested by the previous reviewers. However, there are two things you need to work on to make the console app work properly:

1. When I list the rentals, it shows the rental and then also shows "no records were found for the given ID". For this, you could either use the `find` ruby method to get the particular object and then `puts` it or you remove the else statement. I recommend you use the `find` method. You can read this [article](https://linuxhint.com/ruby-find-elements-in-array/) to learn how to use the `find` method.

![image](https://user-images.githubusercontent.com/77106962/164764953-0e7200e1-a920-4e69-8565-855cff64f128.png)

2. You have a typo in `Peson` instead of `Person`.